Libraries/LibWeb/IndexedDB/Internal/ObjectStore.cpp

```diff
@@ -47,4 +48,23 @@ void ObjectStore::remove_records_in_range(GC::Ref<IDBKeyRange> range)
     });
 }
 
+bool ObjectStore::has_record_with_key(GC::Ref<Key> key)
+{
+    auto index = m_records.find_if([&key](auto const& record) {
+        return record.key == key;
+    });
+
+    return index != m_records.end();
+}
+
+void ObjectStore::store_a_record(Record const& record)
+{
+    m_records.append(record);
+
+    // NOTE: The record is stored in the object store’s list of records such that the list is sorted according to the key of the records in ascending order.
+    AK::quick_sort(m_records, [](auto const& a, auto const& b) {
+        return Key::compare_two_keys(a.key, b.key) < 0;
+    });
+}
```

Comment:
Surely we have a way to 'insert at the correct location'? or at least to sort in a smarter way than a simple quick-sort. If not that's okay I suppose, we'll find out during benchmarks if it really matters.

Reply:
Im not aware of any method that "does the right thing" in this case.

All objects under the `Internal` subfolder should call out to "other sources", so once we get a DB implementation up and running, this sort will go away. This is currently a "memory-only" based implementation.